railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -170,6 +170,11 @@ def options_for_migration
           if reference? && !polymorphic?
             options[:foreign_key] = true
           end
+
+
+          if reference? && defined?(Rails) && Rails.application && Rails.application.config && Rails.application.config.generators && Rails.application.config.generators.active_record && Rails.application.config.generators.active_record.has_key?(:primary_key_type)
```

Comment:
This conditional shows this is the wrong place to do this. This change should be in the migration template here https://github.com/rails/rails/blob/fe4e9d4c5d82806fdaf28c87c731132f81f0ac3e/activerecord/lib/rails/generators/active_record/migration/templates/create_table_migration.rb.tt and https://github.com/rails/rails/blob/fe4e9d4c5d82806fdaf28c87c731132f81f0ac3e/activerecord/lib/rails/generators/active_record/migration/templates/migration.rb.tt.

Reply:
@rafaelfranca but it gets added to those templates as a result of being defined where it currently is. How is it any different than the previous conditional:

https://github.com/rails/rails/pull/32461/files/43fe042b6eb6d9fa2fe916191301fabe6929800a#diff-06ee83dd4b3615868328dd430573327bR170

Reply:
`&& defined?(Rails) && Rails.application && Rails.application.config && Rails.application.config.generators && Rails.application.config.generators.active_record && Rails.application.config.generators.active_record.has_key?(:primary_key_type)`

That is the difference. If you have to check all of that is because the place where your doing your change is the wrong one. The correct place would do that conditional in the template that would already have the [`primary_key_type` set in the generator](https://github.com/rails/rails/blob/fe4e9d4c5d82806fdaf28c87c731132f81f0ac3e/activerecord/lib/rails/generators/active_record/migration/migration_generator.rb#L10).

Reply:
@rafaelfranca thanks so much that makes more sense